x/ibc-transfer/handler.go

```diff
@@ -31,14 +31,17 @@ func handleMsgTransfer(ctx sdk.Context, k keeper.Keeper, msg *types.MsgTransfer)
 
 	k.Logger(ctx).Info("IBC transfer: %s from %s to %s", msg.Amount, msg.Sender, msg.Receiver)
 
-	ctx.EventManager().EmitEvent(
+	ctx.EventManager().EmitEvents(sdk.Events{
```

Comment:
One thing I've seen in the akash codebase that would be really nice here is typed events. See the 
- [event types for a module](https://github.com/ovrclk/akash/blob/master/x/deployment/types/event.go#L24)
- [event creation](https://github.com/ovrclk/akash/blob/master/x/deployment/keeper/keeper.go#L129)
- [consuming the events](https://github.com/jackzampolin/deploy/blob/master/cmd/event-handlers.go#L57)

Its a really nice pattern and allows for really ergonomic go code.

Reply:
oh yea this makes a lot of sense, would prevent inconsistencies in event emission as well. I think this would be a good to do in the next iteration on events. I can open an issue for it